Dispatcher/Repositories/ArticlesRepository.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  ArticlesRepository.swift
+//  Dispatcher
+//
+//  Created by Ofek Cohen on 30/04/2023.
+//
+
+import Foundation
+
+class ArticlesRepository {
+    
+    static let shared = ArticlesRepository()
+    let alamofireManager = AlamofireManager()
+    
+    private init() {}
+
+    func getArticlesFromApi(pageNumber: Int, completionHandler: @escaping (_ articles: [Article], _ totalPages: Int, _ errorMsg: String?) -> Void) {
+        let request = buildRequest(nil, pageNumber)
+        alamofireManager.sendRequest(request) { (result: Result<ArticleApiObject, Error>) in
+            switch result {
+            case .success(let dataResult):
+                var articles: [Article] = []
+                for article in dataResult.articles {
+                    if(article.language == "en") {
+                        articles.append(self.buildArticleFromArticleResponse(article))
+                    }
+                }
+                completionHandler(articles, dataResult.totalPages, nil)
+            case .failure(let error):
+                completionHandler([], 0, error.localizedDescription)
+            }
+        }
+    }
+    
+    func getArticlesByUserSearchWords(_ searchKeywords: String, pageNumber: Int, completionHandler: @escaping ( _ articles: [Article], _ totalPages: Int, _ errorMsg: String?) -> Void) {
+        let request = buildRequest(searchKeywords, pageNumber)
+        alamofireManager.sendRequest(request) { (result: Result<ArticleApiObject, Error>) in
+            switch result {
+            case .success(let dataResult):
+                var articles: [Article] = []
+                for article in dataResult.articles {
+                    if(article.language == "en") {
+                        articles.append(self.buildArticleFromArticleResponse(article))
+                    }
+                }
+               
+                completionHandler(articles, Int(dataResult.totalPages), nil)
+            case .failure(let error):
+                completionHandler([], 0, error.localizedDescription)
+            }
+        }
+    }
+    
+    private func buildRequest(_ searchKeywords: String? , _ pageNumber: Int) -> Request {
+        if(searchKeywords != nil) {
```

Comment:
When you need to use the variable that you are checking inside the { } it's better to use "if let" instead of "!= nil" because then you don't need to force-unwrap the variable using "!".
Even though it won't fail because you indeed checked it's not null, we still don't like to use the "!"